starknet/src/utils/merkle.cairo

```diff
@@ -0,0 +1,74 @@
+use core::traits::Into;
+use array::{ArrayTrait, Span, SpanTrait};
+use option::OptionTrait;
+use serde::Serde;
+use sx::types::UserAddress;
+use clone::Clone;
+use hash::{LegacyHash};
+use debug::PrintTrait;
+
+/// Leaf struct for the merkle tree
+#[derive(Copy, Clone, Drop, Serde)]
+struct Leaf {
+    address: UserAddress,
+    voting_power: u256,
+}
+
+trait Hash<T> {
+    fn hash(self: @T) -> felt252;
+}
+
+impl LegacyHashSpan of LegacyHash<Span<felt252>> {
```

Comment:
also add this impl in stark sig auth pr so can remove from whichever pr we merge second.

Reply:
Idk how we should organize files for this... maybe put it in its own file for easy import?